From 0aeab6ed49a556577099431f604a307b0a23d72e Mon Sep 17 00:00:00 2001
From: Robin Gong <b38343@freescale.com>
Date: Sun, 4 Jan 2015 16:56:11 +0800
Subject: [PATCH 0769/1691] MLK-10073-3 cpufreq: imx6q-cpufreq: add
 arm-soc-shared property

commit 0aeab6ed49a556577099431f604a307b0a23d72e from
git://git.freescale.com/imx/linux-2.6-imx.git imx_4.1.15_1.0.0_ga

Pfuze200 only provide one power supply for VDDARM_IN and VDDSOC_IN,
for ldo-bypass mode, we have to pretend they are different regulators
otherwise regulator famework will refuse update voltage.

Signed-off-by: Robin Gong <b38343@freescale.com>
(cherry picked from commit 1b41ab90288b2accd710f8852c11753007749e4a)
---
 drivers/cpufreq/imx6q-cpufreq.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/drivers/cpufreq/imx6q-cpufreq.c b/drivers/cpufreq/imx6q-cpufreq.c
index 15cb607..a00af7d 100644
--- a/drivers/cpufreq/imx6q-cpufreq.c
+++ b/drivers/cpufreq/imx6q-cpufreq.c
@@ -241,6 +241,16 @@ static int imx6q_cpufreq_probe(struct platform_device *pdev)
 	}
 
 	/*
+	 * soc_reg sync  with arm_reg if arm shares the same regulator
+	 * with soc. Otherwise, regulator common framework will refuse to update
+	 * this consumer's voltage right now while another consumer voltage
+	 * still keep in old one. For example, imx6sx-sdb with pfuze200 in
+	 * ldo-bypass mode.
+	 */
+	of_property_read_u32(np, "fsl,arm-soc-shared", &i);
+	if (i == 1)
+		soc_reg = arm_reg;
+	/*
 	 * We expect an OPP table supplied by platform.
 	 * Just, incase the platform did not supply the OPP
 	 * table, it will try to get it.
-- 
1.9.1

